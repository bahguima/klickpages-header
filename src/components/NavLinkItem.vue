<template>
  <li :class="{ locked: !enabled}">
    <a :href="enabled ? url : 'javascript:;' ">
      <i :class="`${icon} mr`"></i>
      <span class="title">{{linkTitle}}</span>
    </a>
    <div v-if="linkDescription" class="tooltip-status">
      <div>
        <div class="tooltip-inner">
          <i :class="`icon-left ${icon}`"></i>
          {{linkDescription}}
        </div>
        <div class="message-content">
          <i class="icon-left icon-kp-lock"></i>
          <div>
              {{topBarConfig.upgrade.description}}
            <a
              :href="topBarConfig.upgrade.url"
              target="_blank"
              class="btn-round"
            >{{topBarConfig.upgrade.title}}</a>
          </div>
        </div>
      </div>
    </div>
  </li>
</template>

<script>
import { mapGetters } from 'vuex';

export default {
  name: 'NavLinkItem',
  props: {
    url: {
      type: String,
      default: '',
    },
    icon: {
      type: String,
      default: '',
    },
    linkTitle: {
      type: String,
      default: '',
    },
    linkDescription: {
      type: String,
      default: '',
    },
    enabled: {
      type: Boolean,
      default: true,
    },
  },
  computed: mapGetters({
    topBarConfig: 'topBar/config',
  }),
};
</script>
