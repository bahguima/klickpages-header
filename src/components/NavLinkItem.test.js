import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import NavLinkItem from './NavLinkItem.vue';

describe('components/NavLinkItem.vue', () => {
  const localVue = createLocalVue();
  localVue.use(Vuex);

  let wrapper;

  const topBar = {
    namespaced: true,
    getters: {
      config: () => ({
        upgrade: {
          title: '',
          description: '',
          url: '',
        },
      }),
    },
  };

  const propsData = {
    url: 'https://url.com',
    icon: 'icon',
    linkTitle: 'linkTitle',
    linkDescription: 'linkDescription',
    enabled: true,
  };

  describe('when component is mounted', () => {
    describe('and enabled prop is true', () => {
      beforeAll(() => {
        const store = new Vuex.Store({ modules: { topBar } });
        wrapper = shallowMount(NavLinkItem, { localVue, store, propsData });
      });

      it('should renders correctly', () => {
        expect(wrapper.element).toMatchSnapshot();
      });

      it('should not contains locked class', () => {
        expect(wrapper.contains('.locked')).toBe(false);
      });

      it('should contains enabled link', () => {
        expect(wrapper.contains(`[href="${propsData.url}"]`)).toBe(true);
      });

      afterAll(() => {
        wrapper.destroy();
        jest.clearAllMocks();
      });
    });

    describe('and enabled prop is false', () => {
      beforeAll(() => {
        propsData.enabled = false;

        const store = new Vuex.Store({ modules: { topBar } });
        wrapper = shallowMount(NavLinkItem, { localVue, store, propsData });
      });

      it('should renders correctly', () => {
        expect(wrapper.element).toMatchSnapshot();
      });

      it('should contains locked class', () => {
        expect(wrapper.contains('.locked')).toBe(true);
      });

      it('should contains disabled link', () => {
        expect(wrapper.contains('[href="javascript:;"]')).toBe(true);
      });

      afterAll(() => {
        wrapper.destroy();
        jest.clearAllMocks();
      });
    });

    describe('and have linkDescription prop', () => {
      beforeAll(() => {
        const store = new Vuex.Store({ modules: { topBar } });
        wrapper = shallowMount(NavLinkItem, { localVue, store, propsData });
      });

      it('should renders correctly', () => {
        expect(wrapper.element).toMatchSnapshot();
      });

      it('should contains tooltip-status class', () => {
        expect(wrapper.contains('.tooltip-status')).toBe(true);
      });

      afterAll(() => {
        wrapper.destroy();
        jest.clearAllMocks();
      });
    });

    describe('and does not have linkDescription prop', () => {
      beforeAll(() => {
        propsData.linkDescription = '';

        const store = new Vuex.Store({ modules: { topBar } });
        wrapper = shallowMount(NavLinkItem, { localVue, store, propsData });
      });

      it('should renders correctly', () => {
        expect(wrapper.element).toMatchSnapshot();
      });

      it('should not contains tooltip-status class', () => {
        expect(wrapper.contains('.tooltip-status')).toBe(false);
      });

      afterAll(() => {
        wrapper.destroy();
        jest.clearAllMocks();
      });
    });
  });
});
