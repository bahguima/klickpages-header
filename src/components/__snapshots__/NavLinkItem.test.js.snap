// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/NavLinkItem.vue when component is mounted and does not have linkDescription prop should renders correctly 1`] = `
<li
  class="locked"
>
  <a
    href="javascript:;"
  >
    <i
      class="icon mr"
    />
     
    <span
      class="title"
    >
      linkTitle
    </span>
  </a>
   
  <!---->
</li>
`;

exports[`components/NavLinkItem.vue when component is mounted and enabled prop is false should renders correctly 1`] = `
<li
  class="locked"
>
  <a
    href="javascript:;"
  >
    <i
      class="icon mr"
    />
     
    <span
      class="title"
    >
      linkTitle
    </span>
  </a>
   
  <div
    class="tooltip-status"
  >
    <div>
      <div
        class="tooltip-inner"
      >
        <i
          class="icon-left icon"
        />
        
        linkDescription
      
      </div>
       
      <div
        class="message-content"
      >
        <i
          class="icon-left icon-kp-lock"
        />
         
        <div>
          
            
          
          <a
            class="btn-round"
            href=""
            target="_blank"
          >
            
          </a>
        </div>
      </div>
    </div>
  </div>
</li>
`;

exports[`components/NavLinkItem.vue when component is mounted and enabled prop is true should renders correctly 1`] = `
<li
  class=""
>
  <a
    href="https://url.com"
  >
    <i
      class="icon mr"
    />
     
    <span
      class="title"
    >
      linkTitle
    </span>
  </a>
   
  <div
    class="tooltip-status"
  >
    <div>
      <div
        class="tooltip-inner"
      >
        <i
          class="icon-left icon"
        />
        
        linkDescription
      
      </div>
       
      <div
        class="message-content"
      >
        <i
          class="icon-left icon-kp-lock"
        />
         
        <div>
          
            
          
          <a
            class="btn-round"
            href=""
            target="_blank"
          >
            
          </a>
        </div>
      </div>
    </div>
  </div>
</li>
`;

exports[`components/NavLinkItem.vue when component is mounted and have linkDescription prop should renders correctly 1`] = `
<li
  class="locked"
>
  <a
    href="javascript:;"
  >
    <i
      class="icon mr"
    />
     
    <span
      class="title"
    >
      linkTitle
    </span>
  </a>
   
  <div
    class="tooltip-status"
  >
    <div>
      <div
        class="tooltip-inner"
      >
        <i
          class="icon-left icon"
        />
        
        linkDescription
      
      </div>
       
      <div
        class="message-content"
      >
        <i
          class="icon-left icon-kp-lock"
        />
         
        <div>
          
            
          
          <a
            class="btn-round"
            href=""
            target="_blank"
          >
            
          </a>
        </div>
      </div>
    </div>
  </div>
</li>
`;
